flow/activities/maintenance_activity.go

```diff
@@ -90,7 +90,7 @@ func (a *MaintenanceActivity) checkAndWaitIfSnapshot(
 	if mirror.MirrorCreatedAt.AsTime().After(time.Now().Add(-30 * time.Second)) {
 		slog.Info("Mirror was created less than 30 seconds ago, waiting for it to be ready before checking for snapshot",
 			"mirror", mirror.MirrorName, "workflowId", mirror.WorkflowId)
-		time.Sleep(30 * time.Second)
+		time.Sleep(time.Now().Add(35 * time.Second).Sub(mirror.MirrorCreatedAt.AsTime()))
```

Comment:
```suggestion
		time.Sleep(time.Now().Add(35 * time.Second).Sub(mirror.MirrorCreatedAt.AsTime()))
```
Shouldn't we be waiting on `mirror created at` + `30s` - `now` instead?

Should adjust condition too, set a targetTime which is createdAt+30, then check that time is before that, then sleep for difference of those values. Only call `time.Now()` once because logic will be cleaner (for stupid scenarios like slog.Info hangs for a minute)

Reply:
makes sense